x/crosschain/keeper/keeper_cross_chain_tx_prove_outbound_tx.go

```diff
@@ -0,0 +1,195 @@
+
+	if receipt.Status == ethtypes.ReceiptStatusFailed {
+
+	}
+
+	cctx.GetCurrentOutTxParam().OutboundTxHash = txx.Hash().Hex()
+	cctx.CctxStatus.LastUpdateTimestamp = ctx.BlockHeader().Time.Unix()
+
+	// FinalizeOutbound sets final status for a successful vote
+	// FinalizeOutbound updates CCTX Prices and Nonce for a revert
+
+	tmpCtx, commit := ctx.CacheContext()
+	err = func() error { //err = FinalizeOutbound(k, ctx, &cctx, msg, ballot.BallotStatus)
+		cctx.GetCurrentOutTxParam().OutboundTxObservedExternalHeight = uint64(res.Height)
+		oldStatus := cctx.CctxStatus.Status
+		switch receipt.Status {
+		case ethtypes.ReceiptStatusSuccessful:
+			switch oldStatus {
+			case types.CctxStatus_PendingRevert:
+				cctx.CctxStatus.ChangeStatus(types.CctxStatus_Reverted, "")
+			case types.CctxStatus_PendingOutbound:
+				cctx.CctxStatus.ChangeStatus(types.CctxStatus_OutboundMined, "")
+			}
+			newStatus := cctx.CctxStatus.Status.String()
+			err := ctx.EventManager().EmitTypedEvents(&types.EventOutboundSuccess{
+				MsgTypeUrl: sdk.MsgTypeURL(&types.MsgVoteOnObservedOutboundTx{}),
+				CctxIndex:  cctx.Index,
+				//ZetaMinted: msg.ZetaMinted.String(), // FIXME: this is not the correct amount
+				OldStatus: oldStatus.String(),
+				NewStatus: newStatus,
+			})
+			if err != nil {
+				ctx.Logger().Error("Error emitting MsgVoteOnObservedOutboundTx :", err)
+			}
+		case ethtypes.ReceiptStatusFailed:
+			switch oldStatus {
+			case types.CctxStatus_PendingOutbound:
+				// create new OutboundTxParams for the revert
+				cctx.OutboundTxParams = append(cctx.OutboundTxParams, &types.OutboundTxParams{
+					Receiver:           cctx.InboundTxParams.Sender,
+					ReceiverChainId:    cctx.InboundTxParams.SenderChainId,
+					Amount:             cctx.InboundTxParams.Amount,
+					CoinType:           cctx.InboundTxParams.CoinType,
+					OutboundTxGasLimit: cctx.OutboundTxParams[0].OutboundTxGasLimit, // NOTE(pwu): revert gas limit = initial outbound gas limit set by user;
+				})
+				err := k.PayGasInZetaAndUpdateCctx(tmpCtx, cctx.InboundTxParams.SenderChainId, &cctx, false)
+				if err != nil {
+					return err
+				}
+				err = k.UpdateNonce(tmpCtx, cctx.InboundTxParams.SenderChainId, &cctx)
+				if err != nil {
+					return err
+				}
+				cctx.CctxStatus.ChangeStatus(types.CctxStatus_PendingRevert, "Outbound failed, start revert")
+			case types.CctxStatus_PendingRevert:
+				cctx.CctxStatus.ChangeStatus(types.CctxStatus_Aborted, "Outbound failed: revert failed; abort TX")
+			}
+			//newStatus := cctx.CctxStatus.Status.String()
+			//EmitOutboundFailure(ctx, msg, oldStatus.String(), newStatus, cctx) // FIXME: enable this
+		}
+		return nil
+	}()
+	if err != nil {
+		// do not commit tmpCtx
+		cctx.CctxStatus.ChangeStatus(types.CctxStatus_Aborted, err.Error())
+		ctx.Logger().Error(err.Error())
+		k.RemoveOutTxTracker(ctx, chainID, nonce)
+		k.RemoveFromPendingNonces(ctx, tss.TssPubkey, chainID, int64(nonce))
+		k.RemoveOutTxTracker(ctx, chainID, nonce)
```

Comment:
It seems the code that has been copied is not up to date, we removed the double `RemoveOutTxTracker` call